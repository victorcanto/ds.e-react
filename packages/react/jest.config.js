module.exports = {
  roots: ['<rootDir>/src'],
  testRegex: '(/.*\\.test)\\.(ts|tsx)$',
  setupFilesAfterEnv: ['<rootDir>/jest.setup.js'],
  testEnvironment: 'jsdom',
};
