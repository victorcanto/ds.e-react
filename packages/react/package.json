{
  "name": "@ds.e/react",
  "version": "1.0.0",
  "main": "lib/index.js",
  "license": "MIT",
  "devDependencies": {
    "@babel/core": "^7.18.13",
    "@babel/preset-env": "^7.18.10",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
    "@storybook/addon-a11y": "^6.5.10",
    "@storybook/addon-actions": "^6.5.10",
    "@storybook/addon-essentials": "^6.5.10",
    "@storybook/addon-interactions": "^6.5.10",
    "@storybook/addon-knobs": "^6.4.0",
    "@storybook/addon-links": "^6.5.10",
    "@storybook/addon-storysource": "^6.5.10",
    "@storybook/builder-webpack4": "^6.5.10",
    "@storybook/manager-webpack4": "^6.5.10",
    "@storybook/react": "^6.5.10",
    "@storybook/testing-library": "^0.0.13",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.3.0",
    "@types/jest": "^29.0.0",
    "@types/react": "^18.0.17",
    "babel-loader": "^8.2.5",
    "jest": "^29.0.1",
    "jest-environment-jsdom": "^29.0.1",
    "react": "^18.2.0",
    "rollup": "^2.78.1",
    "rollup-plugin-typescript2": "^0.33.0",
    "ts-node": "^10.9.1",
    "typescript": "^4.8.2"
  },
  "scripts": {
    "build": "rollup -c",
    "dev": "yarn build --watch",
    "test": "jest --verbose",
    "test:watch": "yarn test -watch",
    "storybook": "start-storybook -p 6006",
    "build-storybook": "build-storybook"
  },
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {}
}
