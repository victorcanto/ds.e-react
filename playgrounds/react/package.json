{
  "name": "@playgrounds/react",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "devDependencies": {
    "@types/react": "^18.0.17",
    "@types/react-dom": "^18.0.6",
    "parcel": "^2.7.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "typescript": "^4.8.2"
  },
  "scripts": {
    "dev": "parcel src/index.html -p 3000"
  },
  "private": true
}
