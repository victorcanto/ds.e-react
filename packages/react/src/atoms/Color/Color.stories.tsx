import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { text, select } from '@storybook/addon-knobs';
import Color, { ColorProps } from './Color';

import { Spacing } from '@ds.e/foundation';

import '@ds.e/sass/lib/Utilities.css';

export default {
  title: 'Atoms|Color',
} as ComponentMeta<typeof Color>;

const Template: ComponentStory<typeof Color> = (args) => <Color {...args} />;

export const Common = Template.bind({});
Common.args = {
  hexCode: text('HexCode', 'red'),
} as ColorProps;

export const CustomDimensions = Template.bind({});
CustomDimensions.args = {
  hexCode: text('HexCode', 'red'),
  width: select('Width', Object.values(Spacing), 'xxl'),
  height: select('Height', Object.values(Spacing), 'xxl'),
} as ColorProps;
