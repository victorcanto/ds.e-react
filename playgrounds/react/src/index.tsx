import React from 'react';
import { createRoot } from 'react-dom/client';

import { Color } from '@ds.e/react';
import '@ds.e/scss/lib/Utilities.css';

const container = document.getElementById('root');
const root = createRoot(container!);
root.render(<Color hexCode='blue' width='lg' height='lg' />);
