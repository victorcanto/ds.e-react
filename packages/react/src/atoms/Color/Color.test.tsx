import React from 'react';
import { render } from '@testing-library/react';
import Color from './Color';

test('snapshot Color atom', () => {
  const { asFragment } = render(<Color hexCode='red' width='xl' height='xl' />);
  expect(asFragment()).toMatchSnapshot();
});
