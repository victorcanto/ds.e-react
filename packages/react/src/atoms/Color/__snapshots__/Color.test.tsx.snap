// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Color atom 1`] = `
<DocumentFragment>
  <div
    class="dse-width-xl dse-height-xl"
    style="background-color: red;"
  />
</DocumentFragment>
`;
